prov/sockets/src/sock_ep.c

```diff
@@ -1216,7 +1222,77 @@ int sock_srx_ctx(struct fid_domain *domain,
 	return 0;
 }
 
-static void sock_set_fabric_attr(const struct fi_fabric_attr *hint_attr,
+int sock_get_prefix_len(uint32_t net_addr)
+{
+	int count = 0;
+	while (net_addr > 0) {
+		net_addr = net_addr >> 1;
+		count++;
+	}
+	return count;
+}
+
+char *sock_get_fabric_name(struct sockaddr_in *src_addr)
+{
+	int ret;
+        struct ifaddrs *ifaddrs, *ifa;
+	char *fabric_name = NULL;
+	struct in_addr net_in_addr;
+	struct sockaddr_in *host_addr, *net_addr;
+	char netbuf[35];
+	int prefix_len;
+#if HAVE_GETIFADDRS
+	ret = getifaddrs(&ifaddrs);
+	if (!ret) {
+		for (ifa = ifaddrs; ifa != NULL; ifa = ifa->ifa_next) {
+			if (ifa->ifa_addr == NULL || !(ifa->ifa_flags & IFF_UP) ||
+			     (ifa->ifa_addr->sa_family != AF_INET))
+				continue;
+			if (sock_compare_addr((struct sockaddr_in *)ifa->ifa_addr, src_addr)) {
+				host_addr = (struct sockaddr_in *)ifa->ifa_addr;
+				net_addr = (struct sockaddr_in *)ifa->ifa_netmask;
+				// set fabric name to the network_adress in the format of a.b.c.d/e
+				net_in_addr.s_addr = (uint32_t)((uint32_t) host_addr->sin_addr.s_addr
+							& (uint32_t) net_addr->sin_addr.s_addr);
+				inet_ntop(host_addr->sin_family, (void *)&(net_in_addr), netbuf,
+					   sizeof(netbuf));
+				prefix_len = sock_get_prefix_len(net_addr->sin_addr.s_addr);
+				snprintf(netbuf + strlen(netbuf), sizeof(netbuf) - strlen(netbuf),
+					  "%s%d", "/", prefix_len);
+				fabric_name = strdup(netbuf);
+				return fabric_name;
+			}
+                }
+                freeifaddrs(ifaddrs);
+        }
+#endif
+	return fabric_name;
+}
+
+char *sock_get_domain_name(struct sockaddr_in *src_addr)
+{
+	int ret;
+        struct ifaddrs *ifaddrs, *ifa;
+	char *domain_name = NULL;
+#if HAVE_GETIFADDRS
+	ret = getifaddrs(&ifaddrs);
+	if (!ret) {
+		for (ifa = ifaddrs; ifa != NULL; ifa = ifa->ifa_next) {
+			if (ifa->ifa_addr == NULL || !(ifa->ifa_flags & IFF_UP) ||
+			     (ifa->ifa_addr->sa_family != AF_INET))
+				continue;
+			if (sock_compare_addr((struct sockaddr_in *)ifa->ifa_addr, src_addr)) {
+				domain_name = strdup(ifa->ifa_name);
+				return domain_name;
+			}
+                }
+                freeifaddrs(ifaddrs);
+        }
+#endif
```

Comment:
Please move the ifdef code around the entire function.  In this case, you can have one ifdef around both calls, and add an else clauses with functions that just return NULL.